src/main.cpp

```diff
@@ -4422,6 +4512,166 @@ bool AcceptBlock(CBlock& block, CValidationState& state, CBlockIndex** ppindex,
     }
 
     int nHeight = pindex->nHeight;
+    int splitHeight = -1;
+
+    if (isPoS) {
+        LOCK(cs_main);
+
+        const bool isBlockFromFork = pindexPrev != nullptr && !chainActive.Contains(pindexPrev);
+        CTransaction &stakeTxIn = block.vtx[1];
+
+        // ZC started after PoS.
+        // Check for serial double spent on the same block, TODO: Move this to the proper method..
+        if(nHeight >= Params().Zerocoin_StartHeight()) {
+            vector<CBigNum> inBlockSerials;
+            for (CTransaction tx : block.vtx) {
+                for (CTxIn in: tx.vin) {
+                    if (in.scriptSig.IsZerocoinSpend()) {
+                        CoinSpend spend = TxInToZerocoinSpend(in);
+                        // Check for serials double spending in the same block
+                        if (std::find(inBlockSerials.begin(), inBlockSerials.end(), spend.getCoinSerialNumber()) != inBlockSerials.end()) {
+                            return state.DoS(100, error("%s: serial double spent on the same block", __func__));
+                        }
+                        inBlockSerials.push_back(spend.getCoinSerialNumber());
+                    }
+                }
+            }
+            inBlockSerials.clear();
+        }
+
+
+        // Check whether is a fork or not
+        if (isBlockFromFork) {
+
+            // Start at the block we're adding on to
+            CBlockIndex *prev = pindexPrev;
+
+            // Inputs
+            std::vector<CTxIn> pivInputs;
+            std::vector<CTxIn> zPIVInputs;
+
+            for (CTxIn stakeIn : stakeTxIn.vin) {
+                if(stakeIn.scriptSig.IsZerocoinSpend()){
+                    zPIVInputs.push_back(stakeIn);
+                }else{
+                    pivInputs.push_back(stakeIn);
+                }
+            }
+            const bool hasPIVInputs = !pivInputs.empty();
+            const bool hasZPIVInputs = !zPIVInputs.empty();
+
+            vector<CBigNum> vBlockSerials;
+            CBlock bl;
+            // Go backwards on the forked chain up to the split
+            do {
```

Comment:
phore??
The code is completely different..

Reply:
yes. we use almost the exact same code. 

```
while block not in fork:
  readblock
  for transaction in block
    for input from each transaction
      check if incoming block stake matches a previous output
```

Reply:
haha, come back when you do something from your own..

---------

Man, just checked your code, you have an infinite loop on the AcceptBlock() method that will crash any node just receiving a block from a fork??
`last = pindexPrev->pprev;` it's not going backwards, it's returning to the parent! . So `chainActive.Contains(pindexPrev)`will always return true. 

You just screwed your network, congratulations!

Check my code, nothing similar to your stuff.. feel free to copy and port it to your "network" or what left from it.

You are lucky, this is open source :)

Reply:
we solved that, but thanks!

are you kidding me? you copied the comments too. it's very clear the code was taken from our patch. please properly credit us as the author.

Reply:
Seems that you are kidding me.. stop this shit show. You are just embarrassing yourself and your project.

Hope to not have to repeat again how useless and bad your code is.. even worst thinking that it **was merged**..  If your chain is still alive is a miracle. 
You are the author of "how to stall your chain for dummies", yeah awesome. Congrats.

Read my code again, look how different is to yours.. feel free to port it to your blockchain.

This is my last response, need to continue working.
Wish you the bests on your stuff 🖖 .

Reply:
I did read your code. It is different, however it clearly was initially taken from Phore. I can submit a more formal request if needed.

Reply:
@meyer9 please stop spam, Phore's patch is completely different. There is no any reason to include copyright notice.

Reply:
If completely different, why the same comments? It's highly unlikely that 2 people would use the exact same phrase "loop through every input from said block" since usage of "said" is somewhat odd.  Plus there are other comments too. Just seeing insults being hurled back at Julian based on initial request for credit instead of any explanation for this